config/src/test/scala/com/typesafe/config/impl/ConfigBeanFactoryTest.scala

```diff
@@ -178,6 +178,7 @@ class ConfigBeanFactoryTest extends TestUtils {
         assertNotNull(beanConfig)
         assertNotNull(beanConfig.getValueObject)
         assertNull(beanConfig.getValueObject.getOptionalValue)
+        assertNull(beanConfig.getValueObject.getNullableValue)
```

Comment:
maybe we could test the matrix of combinations of
 - bean constructor sets to null vs non-null
 - config explicitly sets to null or simply omits the setting

If only to be sure we don't accidentally change behavior of one of those matrix cells later.